libsplinter/src/service/network/interconnect/error.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug, PartialEq)]
+pub enum ServiceInterconnectError {
+    StartUpError(String),
+}
+
+impl Error for ServiceInterconnectError {}
+
+impl fmt::Display for ServiceInterconnectError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ServiceInterconnectError::StartUpError(msg) => {
+                write!(f, "Unable to start peer interconnect: {}", msg)
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ServiceLookupError(pub String);
```

Comment:
Is there a compelling reason to use just a `String` instead of the context/source pattern we use elsewhere?

Reply:
There's less complexity here. We're not actually making use of the source in meaningful way beyond adding it to the display string.  For example, one can't match on the `Box<dyn Error>` that we've specified, so a library user can't doing anything other than print it out.